extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,213 @@
+				continue;
+			} else {
+				result += static_cast<char>(num);
+			}
+		} else { // named character reference
+			auto it = info.html5_names.find(match);
+			if (it != info.html5_names.end()) {
+				if (!AddBlob(it->second, result)) {
+					throw ConversionException("Cannot convert codepoint of %s", match);
+				}
+				continue;
+			}
+			for (idx_t x = match.length() - 1; x >= 1; --x) {
+				string substr = match.substr(0, x);
+				it = info.html5_names.find(substr);
+				if (it != info.html5_names.end()) {
+					if (!AddBlob(it->second, result)) {
+						throw ConversionException("Cannot convert codepoint of %s to utf8", match);
+					}
+					start_pos -= (match.length() - x);
+					input.set(input_data + start_pos, input_size - start_pos);
+					break;
+				}
+			}
+			if (it == info.html5_names.end()) {
+				result.append('&' + match);
+			}
+		}
+	}
+	input.AppendToString(&result);
+	return result;
+}
+
+static void EscapeInputStr(string_t &input_str, string &escaped, bool input_quote) {
+	const auto str = input_str.GetData();
+	const auto str_size = input_str.GetSize();
+	for (idx_t i = 0; i < str_size; ++i) {
+		char ch = str[i];
+		switch (ch) {
+		case '&':
+			escaped += "&amp;";
+			break;
+		case '<':
+			escaped += "&lt;";
+			break;
+		case '>':
+			escaped += "&gt;";
+			break;
+		case '"':
+			if (input_quote) {
+				escaped += "&quot;";
+			} else {
+				escaped += "\"";
+			}
+			break;
+		case '\'':
+			if (input_quote) {
+				escaped += "&#x27;";
+			} else {
+				escaped += "\'";
+			}
+			break;
+		default:
+			escaped += ch;
+		}
+	}
+}
+
+void INetFunctions::Escape(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto escape_string = [&](string_t &input_str, bool input_quote) {
+		string escaped {""};
+		EscapeInputStr(input_str, escaped, input_quote);
+		return StringVector::AddString(result, escaped);
+	};
+	if (args.ColumnCount() == 1) {
+		UnaryExecutor::Execute<string_t, string_t>(args.data[0], result, args.size(),
+		                                           [&](string_t input_str) { return escape_string(input_str, true); });
+	} else {
+		BinaryExecutor::Execute<string_t, bool, string_t>(
+		    args.data[0], args.data[1], result, args.size(),
+		    [&](string_t input_str, bool input_quote) { return escape_string(input_str, input_quote); });
+	}
+}
+
+void INetFunctions::Unescape(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &func_expr = state.expr.Cast<BoundFunctionExpression>();
+	auto &info = func_expr.bind_info->Cast<RegexpUnescapeBindData>();
+	auto &lstate = ExecuteFunctionState::GetFunctionState(state)->Cast<RegexLocalState>();
+	UnaryExecutor::Execute<string_t, string_t>(args.data[0], result, args.size(), [&](string_t input_st) {
+		// check whether the input contains an ampersand character. If not, return the input value unchanged.
+		if (ContainsFun::Find(const_uchar_ptr_cast(input_st.GetData()), input_st.GetSize(), const_uchar_ptr_cast("&"),
```

Comment:
Similarly here - can we first do a count of how many bytes the result will have, and then allocate the empty string? During this first pass we can also check if there are any replacements made - in which case we can skip the second step.